### PR TITLE
Add options to DateTime field

### PR DESCRIPTION
**Pull request by OddEssay**
This update adds support to Administrate::Field::DateTime.with_options() for formatting and an alternative title for the field.

It also documents the options in docs.

### CODE REVIEW

**Comment by jrguenther** [comment]
:+1: Looks like a good approach to me!

**Comment by davidalejandroaguilar** [comment]
+1

File: docs/customizing_dashboards.md [hunk 1]
```diff
@@ -76,4 +76,12 @@ if the value is stored by the number of cents:
   )
 ```
 
+Or if you want to apply a different datetime format:
+```ruby
```
Comment (nickcharlton) [inline docs/customizing_dashboards.md]: Tiny thing: Could you put a space between the text and the code block?

File: docs/customizing_dashboards.md [hunk 2]
```diff
@@ -76,4 +76,12 @@ if the value is stored by the number of cents:
...
+  date_of_birth: Field::DateTime.with_options(
+    title: "Birthday",
+    format: "%-d %B",
```
Comment (nickcharlton) [inline docs/customizing_dashboards.md]: Is there an out of the box format style we could use that Rails already defines?

File: lib/administrate/field/date_time.rb [hunk 3]
```diff
@@ -3,6 +3,21 @@
...
+      def to_s
+        if data.nil?
+          ""
+        else
+          data.strftime(format_date)
+        end
+      end
+
+      def format_date
+        options[:format] || "%FT%T%:z"
```
Comment (nickcharlton) [inline lib/administrate/field/date_time.rb]: Is there an out of the box format style we could use that Rails already defines?

File: spec/lib/fields/date_time_spec.rb [hunk 4]
```diff
@@ -1,6 +1,53 @@
-require "spec_helper"
```
Comment (nickcharlton) [inline spec/lib/fields/date_time_spec.rb]: Should this be removed?

**Comment by m5o** [comment]
@OddEssay some minor rework is required here, would you please help.

**Comment by mvz** [comment]
I think things `.with_options(format: :short)` now work out-of-the-box.

**Comment by purinkle** [comment]
@OddEssay, is this still a change you want to make?

**Comment by purinkle** [comment]
Closing due to inactivity.

**Pull request closed without merging**